### PR TITLE
FEAT: libRed Red/System binding

### PR DESCRIPTION
**Pull request by Oldes**
Made this just as a curiosity although I think it may be quite useful... currently I have 2 issues:

1. once `redOpen` is called, the output from `print` functions from hosting R/S app stops.
2. it is not possible to define `redRoutine` from R/S function

### CODE REVIEW

**Comment by dockimbel** [comment]
1. That's expected, because they share the same underlying R/S runtime code.
2. Try using a `[cdecl]` in the R/S function passed to `redRoutine`.

**Comment by Oldes** [comment]
1. it is not good... it would make difficult to use libRed loaded in app made in Red/System... isn't there some workaround? I would prefer to have `print` from the *host* then the *lib*, if I could choose.

2. there is more problems... first... it is not allowed to cast function into int-ptr!, but I used intermediate casting which seems to make compiler satisfied. Unfortunately, although the routine is called now, there are strange values coming into it, so the result is wrong... what may be a reason?

**Comment by Oldes** [comment]
Btw... I have libRed.dll made with  `rc "build libred"` - The *stdcall* version is completely wrong. With *cdecl* only the routine input is strange.

**Comment by Oldes** [comment]
@dockimbel the redRoutine is working now as expected with the fix: https://github.com/red/red/commit/67cebfc50f30ab2caad3e057a51b96203ac864e2

**Comment by greggirwin** [comment]
Can someone confirm that this works? If so, I'll get it merged.

**Comment by greggirwin** [comment]
@9214 or @hiiamboris, can you confirm this PR is OK to merge?

**Comment by 9214** [comment]
@Oldes nice experiment! My 2 cents:
* Personally I would remove enums, since they are not used in the code and at least `RedType!` is outdated; the other way around is to update it, but then it needs to stay in sync with other eventual changes.
* I like all the `#define`s and how they convey the intention, but IMO they should use `-` rather than `_`, to follow the naming conventions. That shouldn't cause any name clashing AFAIK, though I can be wrong.
* Unfortunately I can't help you with the testing on macOS and GTK, perhaps someone else can spin up VMs and do some quick checking?

File: Library/Red/libred-test.reds [hunk 1]
```diff
@@ -0,0 +1,74 @@
+Red/System [
+	Title:   "Red/System Red binding"
+	Author:  "Oldes"
+	File: 	 %red.reds
```
Comment (hiiamboris) [inline Library/Red/libred-test.reds:4]: wrong filename

File: Library/Red/libred-test.reds [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	b [red_integer!]
+	return: [red_integer!]
+	/local ca cb
+][
+	ca: redCInt32 a
+	cb: redCInt32 b
+	redPrint redString "my-add called"
+	redPrint a ;@@ prints: integer   ?!
+	redPrint b ;@@ prints: false     ?!
+	printf ["add called! %ld %ld^/" ca cb] ;@@ this is not visible either:/
```
Comment (hiiamboris) [inline Library/Red/libred-test.reds:26]: comments are outdated
I can see number being printed on all platforms
`printf` only works on Mac and Linux though

**Comment by hiiamboris** [comment]
works (except for printf) on all platforms

**Pull request merged**